metadata/snapshot.go

```diff
@@ -623,10 +728,25 @@ func (s *snapshotter) Commit(ctx context.Context, name, key string, opts ...snap
 
 }
 
-func (s *snapshotter) Remove(ctx context.Context, key string) error {
+func (s *snapshotter) Remove(ctx context.Context, key string) (err error) {
 	s.l.RLock()
 	defer s.l.RUnlock()
 
+	defer func() {
+		// Broadcast to the waiters so we can return and try again. The unpacker logic special cases ErrAlreadyExists to handle remote
```

Comment:
Does this mean we'll just retry to pull a layer a total of N times across all waiters, where N is the number of attempts that unpacker will make? If it is, then I think it's definitely better than retrying N times for each waiter.

Reply:
Just about yes, it depends who gets woken up and where they fail. If whatever waiter got woken up fails after the Prepare loop, any failure will have that whole pull fail (like what happens now anyways). If a waiter never won the race to have a turn at attempting the unpack then yes, it will try 3 times.